File: src/Parkla.Web/ClientApp/src/app/core/models/reservation.ts
import { AppUser } from "./app-user";
import { ParkSpace } from "./park-space";

export interface Reservation {
  id: number;
  userId: number;
  user: AppUser;
  spaceId: number;
  space: ParkSpace;
  startTime: Date;
  endTime: Date;
}

File: src/Parkla.Core/Entities/Reservation.cs
namespace Parkla.Core.Entities;

public class Reservation : IEntity {
   public int? Id { get; set; }
    public int? UserId { get; set; }
    public virtual User? User { get; set; }
    public int? SpaceId { get; set; }
    public virtual ParkSpace? Space { get; set; }
    public int? PricingId { get; set; }
    public virtual Pricing? Pricing { get; set; }
    public DateTime? StartTime { get; set; }
    public DateTime? EndTime{ get; set; }
}